app_integrations/apps/onelogin.py

```diff
@@ -124,59 +128,50 @@ def _get_onelogin_events(self):
                 }
             ]
         """
-        # OneLogin API expects the ISO 8601 format: YYYY-MM-DDTHH:MM:SSZ
-        formatted_date = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ')
-        params = {'since': formatted_date}
-
         # Make sure we have authentication headers
         if not self._auth_headers:
             return False
 
-        events = self._get_onelogin_paginated_events(params)
-
-        while self._more_to_poll and self._next_page_url:
-            LOGGER.debug('More events to retrieve for \'%s\': %s', self.type(), self._more_to_poll)
-            pagination = self._get_onelogin_paginated_events(None)
+        # Are we just getting events or getting paginated events?
+        if self._next_page_url:
+            params = None
+        else:
+            # OneLogin API expects the ISO 8601 format: YYYY-MM-DDTHH:MM:SSZ
+            formatted_date = datetime.fromtimestamp(
+                self._last_timestamp).strftime('%Y-%m-%dT%H:%M:%SZ')
+            params = {'since': formatted_date}
 
-            # Add the events to our results, this is equivalent to using events.extend()
-            events += pagination
+        LOGGER.debug('Events to retrieve for \'%s\': %s', self.type(), self._more_to_poll)
+        response = self._make_get_request(self._next_page_url, self._auth_headers, params)
 
-        # Return the list of logs to the caller so they can be send to the batcher
-        return events
+        if not response:
+            events = False
 
-    def _get_onelogin_paginated_events(self, params):
-        """Get events from the API pagination url
-
-        Returns:
-            The same as the method _get_onelogin_events()
-        """
-        response = requests.get(self._next_page_url, headers=self._auth_headers, params=params)
-        if not self._check_http_response(response):
-            return False
-
-        # Extract events from response
-        events = response.json()['data']
-
-        # Do we have a pagination link to follow?
-        self._more_to_poll = len(events) >= self._MAX_EVENTS_LIMIT
-
-        # If we are already retrieved all the events, then set the value to prevent more requests
-        self._next_page_url = response.json()['pagination']['next_link']
+        # Extract events and pagination link, if there is any
+        events = response['data']
+        self._next_page_url = response['pagination']['next_link']
 
+        # Return the list of logs to the caller so they can be send to the batcher
         return events
 
     def required_auth_info(self):
         return {
+            'region':
+                {
+                    'description': ('the region for the OneLogin API. This should be a '
+                                    'string of 2 letters, lowercase or uppercase'),
+                    'format': re.compile(r'^[a-zA-Z]{2}$')
```

Comment:
can we restrict this to `eu` or `us` since those are the only regions OneLogin supports?